src/pages/strategies/forms/PrereqForm.js

```diff
@@ -0,0 +1,154 @@
+import { useState, useContext } from 'react'
+import { Form, useFormik } from 'formik'
+import { Grid, FormControlLabel, Checkbox } from '@mui/material'
+import FormShell from 'src/components/Shared/FormShell'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { RequestsContext } from 'src/providers/RequestsContext'
+
+import { useEffect } from 'react'
+
+import { DocumentReleaseRepository } from 'src/repositories/DocumentReleaseRepository'
+
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+
+const PreReqsForm = ({ labels, editMode, maxAccess, setEditMode, recordId, store, setRefresh }) => {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+
+  const [selectedCodeId, setSelectedCodeId] = useState('')
+
+  const { recordId: stgId } = store
+
+  const [initialValues, setInitialData] = useState({
+    codeId: '',
+    prerequisiteId: '',
+    StrategyId: stgId
+  })
+
+  const validationSchema = yup.object({
+    codeId: yup.number().required(),
+    prerequisiteId: yup.number().required()
+  })
+
+  const formik = useFormik({
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues,
+    validationSchema,
+    onSubmit: values => {
+      postPreReq(values)
+    }
+  })
+
+  const postPreReq = async obj => {
+    const isNewRecord = !obj?.codeId
+
+    try {
+      const res = await postRequest({
+        extension: DocumentReleaseRepository.StrategyPrereq.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (isNewRecord) {
+        toast.success('Record Added Successfully')
+        setInitialData(prevData => ({
```

Comment:
why do we need to setInitialData on submit?

Reply:
we dont